src/panels/lovelace/entity-rows/hui-toggle-entity-row.ts

```diff
@@ -0,0 +1,72 @@
+import { html, LitElement, PropertyDeclarations } from "@polymer/lit-element";
+import { TemplateResult } from "lit-html";
+
+import "../components/hui-generic-entity-row.js";
+import "../../../components/entity/ha-entity-toggle.js";
+
+import computeStateDisplay from "../../../common/entity/compute_state_display.js";
+import { hassLocalizeLitMixin } from "../../../mixins/lit-localize-mixin";
+import { HomeAssistant } from "../../../types.js";
+import { EntityRow, EntityConfig } from "./types.js";
+import { HassEntity } from "home-assistant-js-websocket";
+
+class HuiToggleEntityRow extends hassLocalizeLitMixin(LitElement)
+  implements EntityRow {
+  public hass?: HomeAssistant;
+  private _config?: EntityConfig;
+
+  static get properties(): PropertyDeclarations {
+    return {
+      hass: {},
+      _config: {},
+    };
+  }
+
+  public setConfig(config: EntityConfig): void {
+    if (!config || !config.entity) {
+      throw new Error("Invalid Configuration: 'entity' required");
+    }
+
+    this._config = config;
+  }
+
+  protected render(): TemplateResult {
+    if (!this.hass || !this._config) {
+      return html``;
+    }
+
+    const stateObj = this.hass.states[this._config.entity];

```

Comment:
Should we be checking if this exists? Or is this a case of to even call this element there has to be a state obj?

Reply:
We should check if stateObj is defined. Might want to consider updating the type of `hass.states` to indicate that we don't have a value for every key.

Reply:
Ooh, `hui-generic-entity-row` will render a yello block if the entity doesn't exist, we just need to make sure we don't crash on line 46.